components/canvas/peniko_conversions.rs

```diff
@@ -147,16 +147,14 @@ impl Convert<peniko::Brush> for FillOrStrokeStyle {
                 peniko::Brush::Gradient(gradient)
             },
             Surface(surface_style) => {
-                let data = surface_style
-                    .surface_data
-                    .to_owned()
-                    .to_vec(
-                        Some(SnapshotAlphaMode::Transparent {
-                            premultiplied: false,
-                        }),
-                        Some(SnapshotPixelFormat::RGBA),
-                    )
-                    .0;
+                let mut snapshot = surface_style.surface_data.to_owned();
+                snapshot.transform(
+                    SnapshotAlphaMode::Transparent {
+                        premultiplied: false,
+                    },
+                    SnapshotPixelFormat::RGBA,
+                );
+                let data: Vec<u8> = snapshot.into();
```

Comment:
One thing, why I designed the API this way is that, whenever you access bytes, you specify what alpha/pixel format you expect, so you cannot forget to do tranform (you are forced to think about this). In cases where this is not relevant you could always passed None (but again, function call made you think about that).

While just using into is easier, it does not force us to think what data is actually inside.